test/from-draft3/to-draft4/references-tweak.json

```diff
@@ -0,0 +1,89 @@
+[
+    {
+        "title": "Reference to `type` when it has objects in its array",
+        "from": {
+            "type": [ {} ],
+            "$ref": "#/type/0"
+        },
+        "to": {
+            "$ref": "#/anyOf/0",
+            "anyOf": [
+                {}
+            ]
+        }
+    },
+    {
+        "title": "Reference to `type` when it has objects and string in its array",
+        "from": {
+            "type": [ "string", {} ],
+            "$ref": "#/type/1"
+        },
+        "to": {
+            "anyOf": [
+                { "type": ["string"] },
+                {}
+            ]
+        }
+    },
+    {
+        "title":"Reference to `disallow` when it has objects in its array",
+        "from": {
+            "disallow": [ {} ],
+            "$ref": "#/disallow/0"
+        },
+        "to": {
+            "allOf": [
+                { "not": {} }
+            ],
+            "$ref": "#/allOf/0/not"
+        }
+    },
+    {
+        "title": "Reference to `disallow` when it has objects and string in its array",
+        "from": {
+            "disallow": [ "string", {} ],
+            "$ref": "#/disallow/1"
+        },
+        "to": {
+            "allOf": [
+                {
+                    "not": {}
+                },
+                {
+                    "not": {
+                        "type": [
+                            "string"
+                        ]
+                    }
+                }
+            ],
+            "$ref": "#/allOf/0/not"
+        }
+    },
+    {
+        "title": "Reference to `extends` when its an array",
+        "from": {
+            "extends": [ {}, { "type": "object" } ],
+            "$ref": "#/extends/0"
+        },
+        "to": {
+            "allOf": [
+                {}
+            ],
+            "$ref": "#/allOf/1"
```

Comment:
This is not valid, as `allOf` only has one element?

Reply:
@jviotti  in the draft4: https://json-schema.org/draft-04/schema  It has constaints minItems: 1, so it can have 1 item I think but not 0 items...